src/main/ServiceMethod.js

```diff
@@ -1,260 +1,261 @@
+
+                        total = data && data.total; //XXX: temporary transitional leaving it here. some projects expect it as the third arg in callback below
+
+                        util.executePluginChain(plugins.handler, function (plugin) {
+                            var regex = new RegExp(plugin.statusPattern);
+                            newParams.plugin = plugin;
+                            if (regex.test(status)) {
+                                plugin.fn.call(plugin.scope || plugin, payload, newParams, total);
+                            }
+                        });
+                    },
+
+                    providerProgress = function (event) {
+                        util.executePluginChain(plugins.progress, function (plugin) {
+                            plugin.fn.call(plugin.scope || plugin, event.lengthComputable, event.loaded, event.total);
+                        });
+                    };
+
+                //modify the URL with any plugins
+                util.executePluginChain(plugins.url, function (plugin) {
+                    url = plugin.fn.call(plugin.scope || plugin, url, that);
+                });
+
+                // Use a plugged-in provider if there is one
+                if (plugins.provider && plugins.provider.length > 0) {
+                    provider = plugins.provider[plugins.provider.length - 1];
+                    provider = provider.fn.call(provider.scope || provider, this);
+                }
+
+                // Only execute payload plugins if the method expects a payload.
+                if (provider.httpMethodMap[method].hasPayload) {
+                    // add Content-Type header if there is a payload and it has an enctype.
+                    if (payloadParamDef && payloadParamDef.enctype) {
+                        headers = {"Content-Type": payloadParamDef.enctype};
+                        // it's a regular payload so process it.
+                    } else {
+                        requestPayload = this.processRequest(requestPayload, plugins);
+                    }
+                }
+
+                logger.debug("Calling method [" + this.name + "] with URL: " + url);
+                
+                requestParams = {
+                    url: url,
+                    headers: headers,
+                    payload: requestPayload,
+                    handler: providerHandler,
+                    onprogress: providerProgress,
+                    asynchronous: params.asynchronous,
+                    timeout: timeout || "none",
+                    dontExecute: true
+                };
+                
+                if (jsonpCallbackParam) {
+                    requestParams.jsonpCallbackParam = jsonpCallbackParam;
+                }
+
+                newParams.request = provider[provider.httpMethodMap[method].method](requestParams);
+                newParams.response = that.reader.smd.services[that.name];

```

Comment:
Does this work as expected? This looks like it would return the service method specification from the schema, not the result from the actual invocation.